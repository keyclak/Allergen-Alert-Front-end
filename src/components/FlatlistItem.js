import React, {useState} from 'react'
import { View, Text, StyleSheet, Pressable, Image} from 'react-native';
import { ListItem, Icon } from 'react-native-elements'
import { StyleConstants, Styles, Colors } from '../style';

export default function FlatlistItem(props) {

    function handlePress(location) {
        console.log(location)
    }

    return(
        <ListItem key={props.i} bottomDivider onPress={() => {handlePress(props.item.navTo)}}>
            <Icon name={props.item.icon} color={"#666"}/>
            <ListItem.Content>
            <ListItem.Title style={[Styles.labelText, {color:"#666"}]}>{props.item.title}</ListItem.Title>
            </ListItem.Content>
            <ListItem.Chevron color="#666"/>
        </ListItem>
    )
}