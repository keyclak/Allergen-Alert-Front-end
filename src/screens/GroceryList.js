import React, {useEffect, useState} from 'react'
import {SafeAreaView, ScrollView, View, Text, FlatList, Pressable} from 'react-native'
import { StyleConstants, Styles, Colors } from '../style';
import { useDeleteGrocery, useGetFoodByID, useGetGroceryList, useTogglePurchased } from '../hooks/api';
import { ListItem, CheckBox } from 'react-native-elements'

export default function GroceryList({navigation}) {

    const getGroceryList = useGetGroceryList();
    const getFoodByID = useGetFoodByID(); 
    const deleteGrocery = useDeleteGrocery(); 
    const togglePurchased = useTogglePurchased(); 

    useEffect(() => {
        navigation.addListener('focus', () =>  getGroceryList.background());
    }, [navigation]);

    function onDelete(id) {
        deleteGrocery.execute(id)
            .then(() => getGroceryList.background())
            .catch(e => {});
    }

    function onCheck(food) {
        togglePurchased.execute(food.foodId, !food.purchased)
            .then(() => getGroceryList.background())
            .catch(e => {});    
    }

    return (
        <SafeAreaView style={[Styles.container, {flex: 1, alignItems: 'flex-start', justifyContent:'space-evenly'}]}>
            <View style={{paddingTop: 15}}>
                <FlatList
                    data={getGroceryList.response}
                    renderItem={({ item }) => (
                        <ListItem containerStyle={{backgroundColor: Colors.Background, padding: 2}}>
                            <CheckBox
                                containerStyle={{width: '72%', borderRadius: 15}}
                                // TODO: use item.name
                                title={item.foodId}
                                checked={item.purchased}
                                onIconPress={() => onCheck(item)}
                            />
                            <Pressable 
                                style= {{backgroundColor: Colors.ButtonBackground, padding: 15, borderRadius: 25}}
                                onPress={() => onDelete(item.foodId)}>
                                <Text style={{color: Colors.Accent}}>Delete</Text>
                            </Pressable>
                        </ListItem>
                    )}
                    keyExtractor={item => `${item.foodId}`}
                />
            </View>
        </SafeAreaView>
    )
}