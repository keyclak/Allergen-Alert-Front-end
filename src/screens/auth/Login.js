import React, { useContext, useEffect, useState } from 'react';
import {Image, ImageBackground, View, Text, Pressable, StyleSheet, ActivityIndicator, TextInput } from 'react-native';
import LoadingButton from '../../components/LoadingButton';
import TextLoadingButton from '../../components/TextLoadingButton';
import { StyleConstants, Styles, Colors } from '../../style';
import { useLogin, useDummy } from '../../hooks/api';
import { AuthContext } from '../../context';
import FormTextInput from '../../components/FormTextInput';

export default function Login({navigation}) {
    const context = useContext(AuthContext);

    const [username, setUsername] = useState();
    const [password, setPassword] = useState();

    const login = useLogin(username, password);
    const dummy = useDummy("Hello World");

    useEffect(() => console.log(dummy.response), [dummy.response]);

    function onSubmit() {
        login.execute()
            .then(r => context.logIn(r.token))
            .catch(e => {});
    }

    function onCreateAccount() {
        navigation.navigate('CreateAccount');
    }

    function onForgotPass() {
        navigation.navigate('ForgotPass');
    }

    return (
        <View style={Styles.container}>
            <View style={{width: StyleConstants.FormWidth}}>

                <Text style={Styles.logo}>Allergen Alert</Text>

                <TextInput 
                    placeholder="Username" 
                    placeholderTextColor={Colors.Gray[5]} 
                    style={Styles.formInput}/>

                <TextInput 
                    placeholder="Password" 
                    placeholderTextColor={Colors.Gray[5]} 
                    style={Styles.formInput}/>

                <TextLoadingButton text="Log In" onPress={onSubmit} isLoading={login.loading}/>

                <Pressable style={Styles.plainButton} onPress={onForgotPass}>
                    <Text style={{color: Colors.Gray[6]}}>Forgot Password?</Text>
                </Pressable>

                <Pressable style={Styles.plainButton} onPress={onCreateAccount}>
                    <Text style={{color: Colors.Gray[5]}}>Don't have an account? </Text>
                    <Text style={{color: Colors.Gray[6]}}>Create One</Text>
                </Pressable>

            </View>
        </View>
    );
}