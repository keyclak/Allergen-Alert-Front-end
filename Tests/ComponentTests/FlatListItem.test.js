import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import FlatlistItem from '../../src/components/FlatlistItem';
import { create } from "react-test-renderer";
import { FlatList } from 'react-native-gesture-handler';
import { useNavigation } from '@react-navigation/native';
import renderer from 'react-test-renderer';

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import Login from '../../src/screens/auth/Login';
import GroceryList from '../../src/screens/GroceryList';

const Stack = createStackNavigator();
const MockedNavigator = ({component, params = {}}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MockedScreen"
          component={component}
          params={params}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

describe("FlatListItem Tests", () => {
    test("Renders correctly", () => {
        const props = { i:1, item:{icon: "icon1", title: "title1", navTo:{Login}}};

        const tree = renderer.create(<MockedNavigator component={FlatlistItem}></MockedNavigator>);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    // test("Correctly passes item in", () => {
    //     const item = { icon: "icon1", title: "title1" };

    //     const tree = renderer.create(
    //     <MockedNavigator component={Login}>
    //         <FlatlistItem item={item}/>
    //     </MockedNavigator>);

    //     expect(tree.toJSON().props.item.title).toEqual("title1");
    //     expect(tree.toJSON().props.item.icon).toEqual("icon1");
    // });

    it("accepts products props", () => {
        const props = {i:'1', item:{ icon: "icon1", title: "title1", navTo:{Login}}};

        const wrapper = mount(<FlatlistItem i={props.i} item={props.item}/>);
        expect(wrapper.props().item).toEqual(item);
      });
});