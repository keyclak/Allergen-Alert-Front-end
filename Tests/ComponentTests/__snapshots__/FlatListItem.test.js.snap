// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatListItem Tests Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "display": "flex",
                            "height": "100%",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "bottom": "5%",
                                "color": "#2AB7CA",
                                "fontSize": 45,
                              }
                            }
                          >
                            Allergen Alert
                          </Text>
                          <TextInput
                            allowFontScaling={true}
                            onBlur={[Function]}
                            onChangeText={[Function]}
                            onFocus={[Function]}
                            placeholder="Username"
                            placeholderTextColor="#868e96"
                            rejectResponderTermination={true}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#e9ecef",
                                  "borderColor": "#868e96",
                                  "borderRadius": 15,
                                  "borderWidth": 0,
                                  "color": "#495057",
                                  "fontSize": 16,
                                  "height": 50,
                                  "marginBottom": 15,
                                  "paddingLeft": 15,
                                  "paddingRight": 15,
                                },
                                undefined,
                                false,
                              ]
                            }
                            underlineColorAndroid="transparent"
                          />
                          <TextInput
                            allowFontScaling={true}
                            onBlur={[Function]}
                            onChangeText={[Function]}
                            onFocus={[Function]}
                            placeholder="Password"
                            placeholderTextColor="#868e96"
                            rejectResponderTermination={true}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#e9ecef",
                                  "borderColor": "#868e96",
                                  "borderRadius": 15,
                                  "borderWidth": 0,
                                  "color": "#495057",
                                  "fontSize": 16,
                                  "height": 50,
                                  "marginBottom": 15,
                                  "paddingLeft": 15,
                                  "paddingRight": 15,
                                },
                                undefined,
                                false,
                              ]
                            }
                            underlineColorAndroid="transparent"
                          />
                          <View
                            accessible={true}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "alignContent": "center",
                                  "alignItems": "center",
                                  "backgroundColor": "#2AB7CA",
                                  "borderRadius": 15,
                                  "height": 50,
                                  "justifyContent": "center",
                                  "marginBottom": 15,
                                },
                                false,
                                undefined,
                              ]
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "white",
                                  "fontSize": 16,
                                }
                              }
                            >
                              Log In
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 25,
                                "justifyContent": "center",
                                "marginBottom": 15,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#868e96",
                                }
                              }
                            >
                              Forgot Password?
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 25,
                                "justifyContent": "center",
                                "marginBottom": 15,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#adb5bd",
                                }
                              }
                            >
                              Don't have an account? 
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#868e96",
                                }
                              }
                            >
                              Create One
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
